File: SDKs/ably.io/platform/1.1.0/typescript/src/sdk/models/shared/devicedetails.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { RecipientOutput } from "./recipient";
import { Recipient } from "./recipient";


export enum DeviceDetailsFormFactorEnum {
    Phone = "phone",
    Tablet = "tablet",
    Desktop = "desktop",
    Tv = "tv",
    Watch = "watch",
    Car = "car",
    Embedded = "embedded"
}

export enum DeviceDetailsPlatformEnum {
    Ios = "ios",
    Android = "android"
}

export enum DeviceDetailsPushStateEnum {
    Active = "Active",
    Failing = "Failing",
    Failed = "Failed"
}


export class DeviceDetailsOutput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=clientId" })
  clientId?: string;

  @SpeakeasyMetadata({ data: "json, name=deviceSecret" })
  deviceSecret?: string;

  @SpeakeasyMetadata({ data: "json, name=formFactor" })
  formFactor?: DeviceDetailsFormFactorEnum;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=platform" })
  platform?: DeviceDetailsPlatformEnum;

  @SpeakeasyMetadata({ data: "json, name=push.recipient" })
  pushRecipient?: RecipientOutput;

  @SpeakeasyMetadata({ data: "json, name=push.state" })
  pushState?: DeviceDetailsPushStateEnum;
}


export class DeviceDetailsInput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=clientId, form, name=clientId;" })
  clientId?: string;

  @SpeakeasyMetadata({ data: "json, name=deviceSecret, form, name=deviceSecret;" })
  deviceSecret?: string;

  @SpeakeasyMetadata({ data: "json, name=formFactor, form, name=formFactor;" })
  formFactor?: DeviceDetailsFormFactorEnum;

  @SpeakeasyMetadata({ data: "json, name=id, form, name=id;" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=metadata, form, name=metadata;json=true" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=platform, form, name=platform;" })
  platform?: DeviceDetailsPlatformEnum;

  @SpeakeasyMetadata({ data: "json, name=push.recipient, form, name=push.recipient;" })
  pushRecipient?: Recipient;
}

File: SDKs/ably.io/platform/1.1.0/python/src/sdk/models/shared/devicedetails.py
import dataclasses
from typing import Any,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import recipient as shared_recipient
from ..shared import recipient as shared_recipient

class DeviceDetailsFormFactorEnum(str, Enum):
    PHONE = "phone"
    TABLET = "tablet"
    DESKTOP = "desktop"
    TV = "tv"
    WATCH = "watch"
    CAR = "car"
    EMBEDDED = "embedded"

class DeviceDetailsPlatformEnum(str, Enum):
    IOS = "ios"
    ANDROID = "android"

class DeviceDetailsPushStateEnum(str, Enum):
    ACTIVE = "Active"
    FAILING = "Failing"
    FAILED = "Failed"


@dataclass_json
@dataclasses.dataclass
class DeviceDetailsOutput:
    client_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('clientId') }})
    device_secret: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('deviceSecret') }})
    form_factor: Optional[DeviceDetailsFormFactorEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('formFactor') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    metadata: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }})
    platform: Optional[DeviceDetailsPlatformEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('platform') }})
    push_recipient: Optional[shared_recipient.RecipientOutput] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('push.recipient') }})
    push_state: Optional[DeviceDetailsPushStateEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('push.state') }})
    

@dataclass_json
@dataclasses.dataclass
class DeviceDetailsInput:
    client_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('clientId') }, 'form': { 'field_name': 'clientId' }})
    device_secret: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('deviceSecret') }, 'form': { 'field_name': 'deviceSecret' }})
    form_factor: Optional[DeviceDetailsFormFactorEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('formFactor') }, 'form': { 'field_name': 'formFactor' }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }, 'form': { 'field_name': 'id' }})
    metadata: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }, 'form': { 'field_name': 'metadata' }})
    platform: Optional[DeviceDetailsPlatformEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('platform') }, 'form': { 'field_name': 'platform' }})
    push_recipient: Optional[shared_recipient.Recipient] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('push.recipient') }, 'form': { 'field_name': 'push.recipient' }})
    
